CodeReview of 16 from 2

EliaBedin opened the pull request:
Here's the file for the review.
The introduction header should explain everything, but in case, fell free to PM me.
I wish you good work.

Commit:
Review code

Added the file to be reviewed by Molinaro & Prandini

leopardo-rossi-developer commented on app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java:
```diff
@@ -0,0 +1,123 @@
+     * @author Bedin Elia
+     */
+    private void capturePhoto() {
+        if (mCapturingPicture) return;
```
Be careful because you've never initialize variable _mCapturingPicture_ so when the code is executed the first time it could throw an error you don't want. I suggest you to initialize it to _false_ in the onCreate method or directly after the declaration

mattia1142866 replied:
I agree with rossi.

EliaBedin replied:
Thank you both very much. I agree I should have done that.

leopardo-rossi-developer commented on app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java:
```diff
@@ -0,0 +1,123 @@
+public class CameraActivity extends AppCompatActivity {
+
+    private CameraView camera;
+    private boolean mCapturingPicture;
```
The name of this variable isn't very explanatory. If I were you, I'd call it something like _photoIsCaptured_

EliaBedin replied:
Thanks for your feedback.
Yes, the name isn't as explanatory as it should

leopardo-rossi-developer commented on app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java:
```diff
@@ -0,0 +1,123 @@
+ *  The documentation starts here
+ *      https://github.com/natario1/CameraView#usage
+ */
+
```
Very good explanation of this class' goal! In this way who reads this class can understand what is its purpose even before reading the whole code.

EliaBedin replied:
Thank you, very appreciated!

mattia1142866 reviewed (commented):
the code is clear, the only uncertaintly is the inizialization of mCapturingPicture. I also notice that the code is commented in the right way.

Commit:
Fixed critical aspects pointed in the review

Added variables descriptions.
Changed variable name to make it more intuitive.
Fixed typos and comments

Commit:
Tested changes to the code proposed in the review

Overhaul of the way the app checks if it's ready to take a picture using button.setEnabled().
Changed wording for some comments to make them clearer.

Commit:
Removed if statement

Removed an if statement that was here from an old implementation.

mattia1142866 reviewed (approved):
I read the corrections and the important things seem to be resolved
